relayer/processor/path_processor_internal.go

```diff
@@ -581,33 +581,40 @@ func (pp *PathProcessor) assembleAndSendMessages(
 	}
 	outgoingMessages = append(outgoingMessages, msgUpdateClient)
 
-	var sentPackageMessages []packetIBCMessage
-
 	for _, msg := range messages.packetMessages {
-		var assembleMessage func(context.Context, provider.PacketInfo, string, provider.LatestBlock) (provider.RelayerMessage, error)
+		var packetProof func(context.Context, provider.PacketInfo, provider.LatestBlock) (provider.PacketProof, error)
+		var assembleMessage func(provider.PacketInfo, provider.PacketProof) (provider.RelayerMessage, error)
```

Comment:
Two function variables are fine, but at three I would probably pull out a struct or interface for this.